smallrye-reactive-messaging-pulsar/src/main/java/io/smallrye/reactive/messaging/pulsar/tracing/PulsarTrace.java

```diff
@@ -85,7 +85,7 @@ public Builder withUncompressedPayloadSize(Long uncompressedPayloadSize) {
         }
 
         public Builder withProperties(Map<String, String> properties) {
-            this.properties = properties;
+            this.properties = new HashMap<>(properties);
```

Comment:
Making sure, that the OTL support does not break existing code. Otherwise, it fails, if the metadata are created with immutable map (like Map.of)

Reply:
Should it be a concurrent hash map?

Reply:
We need a null check for the properties parameter. The failing tests are related to that. 
I don't think it needs to be a concurrent map, the map is copied to the trace object and extractors get/set on that copy.

Reply:
Yeah, I've created this PR as a draft, more like to describe an issue. I was not able to build the project with the tests (I've got some local setup issues), I'll correct that and fix the failing test.